psi4/driver/procrouting/dft_funcs/dict_builder.py

```diff
@@ -0,0 +1,347 @@
+                    citation.append(x_func.citation())
+                if x_func.description() not in descr:
+                    descr.append(x_func.description())
+        
+        # Exchange processing - HF part:
+        # x_HF contains zeroes or "use_libxc" params from a GGA above
+        if "x_hf" in functionals[name].keys():
+            x_params = functionals[name]["x_hf"]
+            
+            # if "use_libxc" specified here, fetch parameters. 
+            # Duplicate definition of "use_libxc" caught in check_consistency.
+            if "use_libxc" in x_params.keys():
+                x_name = "XC_" + x_params["use_libxc"]
+                x_HF = core.LibXCFunctional(x_name, restricted).query_libxc("XC_HYB_CAM_COEF")
+            if "alpha" in x_params.keys():
+                sup.set_x_alpha(x_params["alpha"])
+            else:
+                x_params["alpha"] = 0.0
+            if "beta" in x_params.keys():
+                sup.set_x_beta(x_params["beta"])
+            if "omega" in x_params.keys():
+                sup.set_x_omega(x_params["omega"])
+        
+        # LibXC uses different nomenclature: 
+        # we need to shuffle the long and short range contributions around
+        # by default, all 3 are 0.0 - different values are set only if "use_libxc" is specified
+        else:
+            sup.set_x_alpha(x_HF["ALPHA"] + x_HF["BETA"])
+            sup.set_x_beta(x_HF["ALPHA"] - x_HF["BETA"])
+            sup.set_x_omega(x_HF["OMEGA"])
+
+        # Correlation processing - GGA part, generally same as above.
+        if "c_functionals" in functionals[name].keys():
+            c_funcs = functionals[name]["c_functionals"]
+            for c_key in c_funcs.keys():
+                c_name = "XC_" + c_key
+                c_func = core.LibXCFunctional(c_name, restricted)
+                c_params = functionals[name]["c_functionals"][c_key]
+                if "tweak" in c_params.keys():
+                    c_func.set_tweak(c_params["tweak"])
+                if "alpha" in c_params.keys():
+                    c_func.set_alpha(c_params["alpha"])
+                else:
+                    c_func.set_alpha(1.0)
+                sup.add_c_functional(c_func)
+                if c_func.citation() not in citation:
+                    citation.append(c_func.citation())
+                if c_func.description() not in descr:
+                    descr.append(c_func.description())
+        
+        # Correlation processing - MP2 part
+        if "c_mp2" in functionals[name].keys():
+            c_params = functionals[name]["c_mp2"]
+            if "alpha" in c_params.keys():
+                sup.set_c_alpha(c_params["alpha"])
+            else:
+                sup.set_c_alpha(0.0)
+            
+            # The value of alpha is locked to 1.0 C++-side when SCS is detected
+            if "ss" in c_params:
+                sup.set_c_ss_alpha(c_params["ss"])
+                sup.set_c_alpha(1.0)
```

Comment:
I think this is the current behaviour enforced C++-side. However, without setting alpha to 1 here, the SCS are not activated!

Reply:
Hmm, I see that in the C++ code. Is there a better way to handle the double hybrid business? This seems a bit awkward at the moment... happy to make some C++ changes based on your recommendations.

Reply:
I implemented it this way cause you suggested removing all of the `"c_mp2": {"alpha":x}` entries - the alpha has to be set python-side for MP2 correlation to be triggered at all. The current version works for me...